docs/tableconfig_schema.rst

```diff
@@ -0,0 +1,204 @@
+
+* ``segmentPushFrequency`` - to indicate how frequently segments are uploaded.
+
+  * Allowed values - ``daily``, ``hourly``
+
+* ``segmentPushType`` - Indicates the type of push to the table.
+
+  * Allowed values - ``APPEND`` means new data will be pushed and appended to the current data in the table, all realtime tables *must* be ``APPEND``. ``REFRESH`` will refresh the entire dataset contained within the table.
+
+* ``replication`` - Number of replicas of data in a table, used for offline tables only.
+
+  * Allowed values - Positive integers
+
+* ``replicasPerPartition`` - Number of of data in a table, used for offline tables only.
+
+  * Allowed values - Positive integers
+
+* ``schemaName`` - Name of the schema that's been uploaded to the controller
+
+  * Allowed values - String
+
+* Time column - using ``timeColumnName`` and ``timeType``, this must match what's configured in the preceeding schema
+
+  * Allowed values - String, this must match the ``timeFieldSpec`` section in the schema
+
+* Segment assignment strategy - Described more on the page `Customizing Pinot <customizations.html#segment-assignment-strategies>`_
+
+
+.. code-block:: none
+
+    "segmentsConfig": {
+      "retentionTimeUnit": "DAYS",
+      "retentionTimeValue": "5",
+      "segmentPushFrequency": "daily",
+      "segmentPushType": "APPEND",
+      "replication": "3",
+      "replicasPerPartition": "3",
+      "schemaName": "myPinotSchmea",
+      "timeColumnName": "daysSinceEpoch",
+      "timeType": "DAYS",
+      "segmentAssignmentStrategy": "BalanceNumSegmentAssignmentStrategy"
+    },
+
+Table Index Config Section
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``tableIndexConfig`` section has information about how to configure:
+
+* ``invertedIndexColumns`` - Indicates a list of real column names as specified in the schema to create inverted indexes for. More info on indexes can be found on the `Index Techniques <index_techniques.html>`_ page.
+
+  * Allowed values - String; string must match the column name in the corresponding schema
+
+* ``noDictionaryColumns`` - Indicates a list of real column names as specified in the schema. Column names present will **not** have a dictionary created. More info on indexes can be found on the `Index Techniques <index_techniques.html>`_ page.
+
+  * Allowed values - String; string must match the column name in the corresponding schema
+
+* ``sortedColumn`` - Indicates a list of real column names as specified in the schema. Data should be sorted based on the column names provided. More info on indexes can be found on the `Index Techniques <index_techniques.html>`_ page.
+
+  * Allowed values - String; string must match the column name in the corresponding schema
+
+* ``aggregateMetrics`` - Switch for the aggregate metrics feature. This feature will aggregate realtime stream data as it is consumed, where applicable, in order to reduce segment sizes. This feature is only available on REALTIME tables.
```

Comment:
useful to describe the feature here? We sum the metric column values of all rows that have the same value for dimension columns and create one row in a realtime segment for all such rows. Alternatively, we can add this as a TODO in Tuning section. Can you capture it as a TODO comment in the tuning section

Reply:
I have a bunch of additional sections captured in my notes, I need to create Jira tickets for these all still. 

I am not sure how add TODOs into  sphinx, I can file jira tickets for all these today. Please resolve is that sounds ok.

Reply:
I've added your sentence to the description also. 

I want to wait for you to resolve this based on the TODO/Jira ticket comments.

Reply:
You can use something like in  PR 3884 (it will appear as a TODO) or like PR 3761 (I think this stays as a comment in the doc and does not show up in real doc)

Reply:
Done. Internal jira ticket has also been created.